rs/tests/testing_verification/testnets/small_with_api_bn.rs

```diff
@@ -0,0 +1,83 @@
+//
+// You can setup this testnet with a lifetime of 180 mins by executing the following commands:
+//
+//   $ ./gitlab-ci/tools/docker-run
+//   $ ict testnet create small --lifetime-mins=180 --output-dir=./small -- --test_tmpdir=./small
+//
+// The --output-dir=./small will store the debug output of the test driver in the specified directory.
+// The --test_tmpdir=./small will store the remaining test output in the specified directory.
+// This is useful to have access to in case you need to SSH into an IC node for example like:
+//
+//   $ ssh -i small/_tmp/*/setup/ssh/authorized_priv_keys/admin admin@
+//
+// Note that you can get the  address of the IC node from the ict console output:
+//
+//   {
+//     nodes: [
+//       {
+//         id: y4g5e-dpl4n-swwhv-la7ec-32ngk-w7f3f-pr5bt-kqw67-2lmfy-agipc-zae,
+//         ipv6: 2a0b:21c0:4003:2:5034:46ff:fe3c:e76f
+//       }
+//     ],
+//     subnet_id: 5hv4k-srndq-xgw53-r6ldt-wtv4x-6xvbj-6lvpf-sbu5n-sqied-63bgv-eqe,
+//     subnet_type: application
+//   },
+//
+// To get access to P8s and Grafana look for the following lines in the ict console output:
+//
+//     prometheus: Prometheus Web UI at http://prometheus.small--1692597750709.testnet.farm.dfinity.systems,
+//     grafana: Grafana at http://grafana.small--1692597750709.testnet.farm.dfinity.systems,
+//     progress_clock: IC Progress Clock at http://grafana.small--1692597750709.testnet.farm.dfinity.systems/d/ic-progress-clock/ic-progress-clock?refresh=10su0026from=now-5mu0026to=now,
+//
+// Happy testing!
+
+use anyhow::Result;
+
+use ic_consensus_system_test_utils::rw_message::install_nns_with_customizations_and_check_progress;
+use ic_registry_subnet_type::SubnetType;
+use ic_system_test_driver::driver::{
+    boundary_node::BoundaryNode,
+    group::SystemTestGroup,
+    ic::{InternetComputer, Node, Subnet},
+    prometheus_vm::{HasPrometheus, PrometheusVm},
+    test_env::TestEnv,
+    test_env_api::{await_boundary_node_healthy, HasTopologySnapshot, NnsCustomizations},
+};
+
+const BOUNDARY_NODE_NAME: &str = "boundary-node-1";
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup)
+        .execute_from_args()?;
+    Ok(())
+}
+
+pub fn setup(env: TestEnv) {
+    PrometheusVm::default()
+        .start(&env)
+        .expect("Failed to start prometheus VM");
+    InternetComputer::new()
+        .add_subnet(Subnet::new(SubnetType::System).add_nodes(1))
+        .add_subnet(Subnet::new(SubnetType::Application).add_nodes(1))
+        .with_unassigned_node(Node::new().with_domain("api-bn.icp.app".to_string()))
+        .with_api_boundary_nodes(1)
+        .setup_and_start(&env)
+        .expect("Failed to setup IC under test");
+    install_nns_with_customizations_and_check_progress(
+        env.topology_snapshot(),
+        NnsCustomizations::default(),
+    );
+    BoundaryNode::new(String::from(BOUNDARY_NODE_NAME))
```

Comment:
Wondering why we need both: `BoundaryNode::new` in addition to `with_api_boundary_nodes`?

Reply:
They are not the same: 
* API BNs only serve the API endpoints (query, call, read_state, status)
* BNs combine the HTTP gateway (serving frontend) and the API BN
At the moment, we don't have the HTTP gateway yet and therefore, we keep the BN such that you can access frontends on a testnet. As we transition fully to the new BN architecture, we can remove the BN and add the HTTP gateway.

Reply:
SG, thanks for the clarification!

Would it make sense to add it to the preamble comment? Currently it only refers to the regular BN, I think.

Reply:
Good point! I did not even look at that description 😞 . I have updated it.